typescript/api/services/FormsService.ts

```diff
@@ -157,10 +162,25 @@ export module Services {
       });
     }
 
-    public getForm = (branding, recordType, editMode, starting: boolean): Observable<FormModel> => {
+    public getForm(branding: BrandingModel, formParam: string, editMode: boolean, recordType: string, currentRec: any): Observable<FormModel> {
+
+      // allow client to set the form name to use
+      const formName = _.isUndefined(formParam) || _.isEmpty(formParam) ? currentRec.metaMetadata.form : formParam;
+
+      if(formName == 'generated-view-only') {
+        return this.generateFormFromSchema(branding, recordType, currentRec);
```

Comment:
Ideally this method would return a promise too but for the moment if the generateFormFromSchema is converted to async/await this can be changed to
`return Observable.of( this.generateFormFromSchema(branding, recordType, currentRec);`